### ISSUE TITLE
watch delay 1min after upgrade k8s.io versions

### ISSUE THREAD

**Issue by snowplayfire** [kubernetes-sigs/kubebuilder#1979]
I use kubebuilder v2.3.1 and controller-gen v0.4.0 to generate controller, the controller works well.  The generated go.mod as follows:
```
require (
	github.com/go-logr/logr v0.1.0
	github.com/onsi/ginkgo v1.11.0
	github.com/onsi/gomega v1.8.1
	k8s.io/apimachinery v0.17.2
	k8s.io/client-go v0.17.2
	sigs.k8s.io/controller-runtime v0.5.0
)
```
Then I upgrade k8s.io/apimachinery and k8s.io/client-go from v1.17 to v1.19, sigs.k8s.io/controller-runtime from v0.5 to v0.6,  watch delay about 1min. 

/kind bug

**Comment by Adirio** [comment]
We are currently using the following versions for the v2 plugin in master (the v2 plugin is the behavior you have by default in v2.3.1):
https://github.com/kubernetes-sigs/kubebuilder/blob/14b27b4e24ba476b19de776f7a05fc56edf993cd/testdata/project-v2/go.mod#L9-L11

Could you check if that version combination works for you?

If you need v1.19, v3 plugin uses the following versions but the v3 plugin is still not available on any stable kubebuilder release yet and the migration guide from v2 to v3 plugin is still not finished.
https://github.com/kubernetes-sigs/kubebuilder/blob/14b27b4e24ba476b19de776f7a05fc56edf993cd/testdata/project-v3/go.mod#L10-L12

**Comment by snowplayfire** [comment]
@Adirio  Thanks for your explanation.

v0.18.6 + v0.6.4,  the version combination doesn't work for me.

You said that the v2 plugin is the behavior I have by default in v2.3.1,  but v0.17.2 + v0.5.0 seems normal.

**Comment by Adirio** [comment]
> @Adirio Thanks for your explanation.
> 
> v0.18.6 + v0.6.4, the version combination doesn't work for me.

IIRC, k8s 1.19 introduced some breaking changes that made controller-runtime v0.7.0 have breaking changes and that's the reason that the v2 project has never been updated as all breaking changes are being applied to v3 plugin.

> You said that the v2 plugin is the behavior I have by default in v2.3.1, can you tell me the reason? v0.17.8 + v0.5.11 seems normal.

With kubebuilders v3.0.0 release, you will still be able to scaffold v2 projects (internally this is done with plugins and thats why I used that term but it is an implementation detailed I should have omitted). So what I meant is that master branch for v2 projects uses the versions I wrote above. v3 projects are still not stable (getting there) and require to migrate your project (migration instructions under construction) but they use the second set of versions.
